cmd/utils/flags.go

```diff
@@ -1394,7 +1408,16 @@ func SetKlayConfig(ctx *cli.Context, stack *node.Node, cfg *cn.Config) {
 	cfg.SenderTxHashIndexing = ctx.GlobalIsSet(SenderTxHashIndexingFlag.Name)
 	cfg.ParallelDBWrite = !ctx.GlobalIsSet(NoParallelDBWriteFlag.Name)
 	cfg.StateDBCaching = ctx.GlobalIsSet(StateDBCachingFlag.Name)
-	cfg.TrieCacheLimit = ctx.GlobalInt(TrieCacheLimitFlag.Name)
+	cfg.TrieNodeCacheConfig = statedb.TrieNodeCacheConfig{
+		CacheType: statedb.TrieNodeCacheType(ctx.GlobalString(TrieNodeCacheTypeFlag.
+			Name)).ToValid(),
+		FastCacheSizeMB:    ctx.GlobalInt(TrieNodeCacheLimitFlag.Name),
+		RedisEndpoints:     ctx.GlobalStringSlice(TrieNodeCacheRedisEndpointsFlag.Name),
+		RedisClusterEnable: ctx.GlobalBool(TrieNodeCacheRedisClusterFlag.Name),
+	}
+	if cfg.TrieNodeCacheConfig.CacheType == "" {
+		logger.Warn("Invalid cache type", "ValidTypes", statedb.ValidTrieNodeCacheTypes)
```

Comment:
```suggestion
		logger.Warn("Invalid cache type", "inType", ctx.GlobalString(TrieNodeCacheTypeFlag.Name), "ValidTypes", statedb.ValidTrieNodeCacheTypes)
```

Reply:
The log message is moved into `ToValid` function and I applied your change also.